File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/OnlineUserChange.java
package tech.zolhungaj.amqapi.servercommands.globalstate;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandTypeOld;

public record OnlineUserChange(
    @Json(name = "name")
    String username,
    Boolean online
) implements Command {
    @Override
    public String commandName() {
        return CommandTypeOld.ONLINE_USER_CHANGE.commandName;
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/OnlineUserChange.kt
package tech.zolhungaj.amqapi.servercommands.globalstate;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.CommandType;

@CommandType("online user change")
public record OnlineUserChange(
    @Json(name = "name")
    String username,
    Boolean online
){}